### PR TITLE
Remove the most outdated and useless parts of p2p/README.md

### PR DESCRIPTION
**Pull request by ImplOfAnImpl**

### CODE REVIEW

File: p2p/README.md [hunk 1]
```diff
@@ -1,6 +1,6 @@
 # P2P subsystem
 
-TODO
+TODO: this document is very incomplete.
```
Comment (ImplOfAnImpl) [inline p2p/README.md:4]: Do we need a separate task/issue for this?

Comment (TheQuantumPhysicist) [reply p2p/README.md:4]: Sure... why not.

**Review by TheQuantumPhysicist — APPROVED** [review]

File: p2p/README.md [hunk 2]
```diff
@@ -40,102 +30,36 @@ Research: TCP hole punching with the help of bootnodes to allow inbound connecti
...
+In addition to maintaining each connection individually, the Mintlayer node will also maintain the overall set of connections by accepting incoming connections/establishing new outbound connections to maintain the total amount of connections at the specified number. This means that the event loop of the node will reject new connections if the number of active connections is higher than it should be; if the number of connections is lower than the lower bound for number of active connections, the node will try establishing new connections with peers it has learned from previous connections.
 
 ### Message exchange
```
Comment (TheQuantumPhysicist) [inline p2p/README.md:49]: Kind of excessive to erase just everything... but I don't care right now.

Some of the frames below are correct.

Comment (ImplOfAnImpl) [reply p2p/README.md:51]: I've rewritten and added those sections back.

**Pull request merged**